crates/sui-core/src/quorum_driver/mod.rs

```diff
@@ -857,8 +858,11 @@ fn is_error_retryable(error: &SuiError) -> bool {
         SuiError::ExecutionError(..) => false,
         SuiError::ByzantineAuthoritySuspicion { .. } => false,
         SuiError::QuorumFailedToGetEffectsQuorumWhenProcessingTransaction { .. } => false,
+        SuiError::ObjectVersionUnavailableForConsumption { .. } => false,
+        SuiError::GasBudgetTooHigh { .. } => false,
+        SuiError::GasBudgetTooLow { .. } => false,
         other => {
-            debug!("uncategorized tx error: {other}");
+            debug!(?tx_digest, "uncategorized tx error: {other}");
```

Comment:
I would suggest increasing the log level to ERROR (we should notice when this happens and fix it)

Also, can we default to true instead? No harm in retrying something that is guaranteed to fail, but not retrying something that can be retried can contribute to objects getting locked.

Reply:
One clarifying note:  When we retry a transaction that is guaranteed to fail, it hides the nature of the bug -- we saw an example of this in v0.21, where if folks tried to publish a package with missing dependencies, instead of failing with an error that explained what was happening, we failed with a timeout because we kept retrying the transaction.

That's not to say that failing on the first try is the best option -- maybe we can do something in the middle, i.e. default to retryable, but preserve past error(s) and share those, rather than a generic timeout?

Reply:
will update to use `ERROR`.

wrt defaulting to true, my thinking is, essentially there is only a limited set of errors that are forgivable, especially when the system matures enough. So in the future we will see a lot of user errors that are not retryable. @amnn gave a good explanation of why defaulting to true is not good for them - the tx will be "suspended" for minutes before announced failed.

I think the most ideal case is, we capture all truly retryable errors, the benefits are:
1. it should be more manageable, compared to capturing all non-retryable errors
2. it gives users signals immediately because it fails fast.

The ideal case may take some time though, I've been watching these logs since last devnet release. And the PR added the uncategorized errors that we saw, which are not a lot

Reply:
fair enough. Eventually I hope our ERROR logs are clean enough to investigate directly.